types/time.go

```diff
@@ -1664,6 +1664,7 @@ func parseSingleTimeValue(unit string, format string, strictCheck bool) (int64,
 		}
 		dayCount := iv / int64(GoDurationDay/gotime.Second)
 		iv %= int64(GoDurationDay / gotime.Second)
+		dv *= sign
```

Comment:
Should we put this line into `if decimalPointPos < lf {` section?

Reply:
OK, It's much better.